### PR TITLE
USHIFT-1900: Refine run_if_changed regex

### PR DESCRIPTION
**Pull request by pacevedom**

### CODE REVIEW

File: ci-operator/config/openshift/microshift/openshift-microshift-main.yaml [hunk 1]
```diff
@@ -72,18 +72,18 @@ tests:
...
-  run_if_changed: ^(assets/.*|scripts/auto-rebase/.*|go\.mod)
+  run_if_changed: ^(assets/.*|scripts/auto-rebase/.*|go\.mod)$
   steps:
     env:
       DRY_RUN: "y"
     workflow: openshift-microshift-rebase
 - as: microshift-metal-tests
   cluster: build02
-  run_if_changed: ^(assets/.*|cmd/.*|etcd/.*|pkg/.*|packaging/.*|test/.*|vendor/.*|go\.sum|go\.mod|Makefile|.*\.sh|.*\.ks\.template)
+  run_if_changed: ^(assets/.*|cmd/.*|etcd/.*|pkg/.*|packaging/.*|test/.*|vendor/.*|Makefile.*|scripts/.*|.*\.ks\.template)$
```
Comment (ggiguash) [inline ci-operator/config/openshift/microshift/openshift-microshift-main.yaml:86]: Do we need the `.ks.template` part? It is covered by `test/.*`, right?

File: ci-operator/config/openshift/microshift/openshift-microshift-main.yaml [hunk 2]
```diff
@@ -130,7 +130,7 @@ tests:
     workflow: openshift-microshift-rebase
 - as: metal-periodic-test
   cluster: build02
-  run_if_changed: (.*periodic.*|^test/bin/.*)
+  run_if_changed: ^(assets/.*|cmd/.*|etcd/.*|pkg/.*|packaging/.*|test/.*|vendor/.*|Makefile.*|scripts/.*|.*\.ks\.template)$
```
Comment (ggiguash) [inline ci-operator/config/openshift/microshift/openshift-microshift-main.yaml:133]: Same here about the `ks.template`

File: ci-operator/config/openshift/microshift/openshift-microshift-main.yaml (hunk 1 above)
Comment (pacevedom) [reply ci-operator/config/openshift/microshift/openshift-microshift-main.yaml]: Done

File: ci-operator/config/openshift/microshift/openshift-microshift-main.yaml (hunk 2 above)
Comment (pacevedom) [reply ci-operator/config/openshift/microshift/openshift-microshift-main.yaml]: Done

**Pull request merged**